File: gpt/python/120.py

import binascii
salt_bytes = binascii.unhexlify('f7fe4d511bcd33321747a778dd21097f4c0ff98f1e0eba39')
hash = crypto.pbkdf2('test', salt_bytes, 1000, 24, hashlib.sha1)

File: mixtral/javascript/120.js
const co = require('co');
const request = require('request');
const crypto = require('crypto');

const computeHash = co.wrap(function* main(url) {
  return new Promise((resolve, reject) => {
    const req = request({ url });
    const hash = crypto.createHash('sha1');

    // Pipe the response directly into the hash function
    req
      .on('error', (err) => {
        console.log(`Error: ${err}`);
        reject(err);
      })
      .pipe(hash)
      .setEncoding('binary')
      .once('finish', () => resolve(hash.read().toString('hex')));
  });
});

// Test the function
computeHash('http://example.com')
  .then(hash => console.log(`Hash: ${hash}`))
  .catch(err => console.log(`Error: ${err}`));